src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -83,7 +84,7 @@ private function evaluate(JsonPath $query): array
             return $this->evaluateTokensOnDecodedData($tokens, $data);
         } catch (InvalidArgumentException $e) {
             throw $e;
-        } catch (\Throwable $e) {
+        } catch (InvalidJsonPathException $e) {
```

Comment:
catching Throwable is a terrible practice that needs great care
here it's certainly not justified